tests/Exception/UnknownFileTypeExceptionTest.php

```diff
@@ -7,11 +7,27 @@
 
 final class UnknownFileTypeExceptionTest extends PHPUnit_Framework_TestCase
 {
-    public function testItIsAnInstanceOfThePackagesRuntimeException()
+    public function testItImplementsTheBaseExceptionType()
     {
         $this->assertInstanceOf(
-            'TomPHP\ConfigServiceProvider\Exception\RuntimeException',
+            'TomPHP\ConfigServiceProvider\Exception\Exception',
             new UnknownFileTypeException()
         );
     }
+
+    public function testItIsADomainException()
+    {
+        $this->assertInstanceOf('DomainException', new UnknownFileTypeException());
+    }
+
+    public function testItCanBeCreatedFromFileExtension()
+    {
+        $this->assertEquals(
+            'No reader configured for ".yml" files; readers are available for [".json", ".php"].',
+            UnknownFileTypeException::fromFileExtension(
+                '.yml',
+                ['.json', '.php']
+            )->getMessage()
+        );
```

Comment:
@tomphp 

The wrapping seems a bit off, maybe worth extracting a variable?

```php
$exception = UnknownFileTypeException::fromFileExtension(
    '.yml',
    ['.json', '.php']
);

$this->assertEquals(
    'No reader configured for ".yml" files; readers are available for [".json", ".php"].', 
    $exception->getMessage()
);
```

?

Not sure what your stance is in regard to assertions, I usually try to be as strict as possible - that is, unless I really don't care about the returned type, I'd always assert sameness - what do you think?